docs/reference/execution-providers/CUDA-ExecutionProvider.md

```diff
@@ -0,0 +1,80 @@
+---
+title: CUDA
+parent: Execution Providers
+grand_parent: Reference
+nav_order: 1
+---
+
+# CUDA Execution Provider
+
+The CUDA Execution Provider enables hardware accelerated computation on Nvidia CUDA-enabled GPUs.
+
+## Build
+For build instructions, please see the [BUILD page](../../how-to/build.md#CUDA).
+
+## Configuration Options
+The CUDA Execution Provider supports the following configuration options.
+
+### device_id
+The device ID.
+
+### cuda_mem_limit
+The size limit of the device memory arena in bytes. This size limit is only for the execution provider's arena. The total device memory usage may be higher.
+
+### arena_extend_strategy
+The strategy for extending the device memory arena.
+
+Value                   | Description
+-|-
+kNextPowerOfTwo (0)     | subsequent extensions extend by larger amounts (multiplied by powers of two)
+kSameAsRequested (1)    | extend by the requested amount
+
+### cudnn_conv_algo_search
+The type of search done for cuDNN convolution algorithms.
+
+Value           | Description
+-|-
+EXHAUSTIVE (0)  | expensive exhaustive benchmarking using cudnnFindConvolutionForwardAlgorithmEx
+HEURISTIC (1)   | lightweight heuristic based search using cudnnGetConvolutionForwardAlgorithm_v7
+DEFAULT (2)     | default algorithm using CUDNN_CONVOLUTION_FWD_ALGO_IMPLICIT_PRECOMP_GEMM
+
+### do_copy_in_default_stream
+Whether to do copies in the default stream or use separate streams. The recommended setting is true. If false, there are race conditions and possibly better performance.
+
+## Example Usage
+
+### Python
+
+```python
+import onnxruntime as ort
+
+model_path = '<path to model>'
+
+providers = [
+    ('CUDAExecutionProvider', {
+        'device_id': 0,
+        'arena_extend_strategy': 'kNextPowerOfTwo',
+        'cuda_mem_limit': 2 * 1024 * 1024 * 1024,
+        'cudnn_conv_algo_search': 'EXHAUSTIVE',
```

Comment:
I think the key you want in the example is `cudnn_conv_algo`. reading the code, that is the key it is going to be looking for

Reply:
it was just updated by this PR: [https://github.com/microsoft/onnxruntime/pull/6171](https://github.com/microsoft/onnxruntime/pull/6171)

---
In reply to: [553438408](https://github.com/microsoft/onnxruntime/pull/6253#discussion_r553438408) [](ancestors = 553438408)

Reply:
Ok